Makefile

```diff
@@ -12,20 +12,27 @@ build:
 TEST_FILE ?= tests/unit_tests
 
 # Define a variable for Python and notebook files.
-lint format: PYTHON_FILES=.
-lint_diff format_diff: PYTHON_FILES=$(shell git diff --relative=. --name-only --diff-filter=d master | grep -E '\.py$$|\.ipynb$$')
+PYTHON_FILES=.
 
-lint lint_diff:
-	[ "$(PYTHON_FILES)" = "" ] ||	uv run ruff format $(PYTHON_FILES) --diff
-	[ "$(PYTHON_FILES)" = "" ] ||	uv run ruff check $(PYTHON_FILES) --diff
-	[ "$(PYTHON_FILES)" = "" ] || uv run mypy $(PYTHON_FILES)
+lint: build
```

Comment:
The changes in this file don't look correct..

the main `lint` is for linting python code and it shouldn't depend on `build`

`lint` and `lint-ci` should be one and the same -- we don't want a differentiation between the linting that runs locally vs. the linting that runs remotely

`format` should also not depend on build